src/main/kotlin/snyk/oss/annotator/OSSBaseAnnotator.kt

```diff
@@ -0,0 +1,131 @@
+package snyk.oss.annotator
+
+import com.intellij.lang.annotation.AnnotationBuilder
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.ExternalAnnotator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiFile
+import io.snyk.plugin.getSnykToolWindowPanel
+import snyk.common.intentionactions.AlwaysAvailableReplacementIntentionAction
+import snyk.oss.OssVulnerabilitiesForFile
+import snyk.oss.Vulnerability
+import snyk.oss.annotator.AnnotatorHelper.severity
+import kotlin.math.max
+
+abstract class OSSBaseAnnotator : ExternalAnnotator<PsiFile, Unit>() {
+    // overrides needed for the Annotator to invoke apply(). We don't do anything here
+    override fun collectInformation(file: PsiFile): PsiFile = file
+    override fun doAnnotate(collectedInfo: PsiFile?): Unit = Unit
+
+    override fun apply(psiFile: PsiFile, annotationResult: Unit, holder: AnnotationHolder) {
+        val issues = getIssuesForFile(psiFile)
+        val remediation = issues?.remediation
```

Comment:
?: return

Reply:
This would not be correct - even if there is no remediation, annotations should occur.